src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -391,42 +391,79 @@ public Set<DatasetDTO> describeDatasets(Integer userId) {
         List<Integer> datasetIdsInUse = dataAccessRequestDAO.findAllDARDatasetRelationDatasetIds();
         HashSet<DatasetDTO> datasets = new HashSet<>();
         if (userHasRole(UserRoles.ADMIN.getRoleName(), userId)) {
-            datasets.addAll(datasetDAO.findAllDatasets());
+            datasets.addAll(datasetDAO.findAllDatasetDTOs());
         } else {
-            datasets.addAll(datasetDAO.findActiveDatasets());
+            datasets.addAll(datasetDAO.findActiveDatasetDTOs());
             if (userHasRole(UserRoles.CHAIRPERSON.getRoleName(), userId)) {
-                Collection<DatasetDTO> chairSpecificDatasets = datasetDAO.findDatasetsByUserId(userId);
+                Collection<DatasetDTO> chairSpecificDatasets = datasetDAO.findDatasetDTOsByUserId(userId);
                 datasets.addAll(chairSpecificDatasets);
             }
         }
         datasets.forEach(d -> d.setDeletable(!datasetIdsInUse.contains(d.getDataSetId())));
         return datasets;
     }
 
+
+    public List<Dataset> searchDatasets(String query, User user) {
+        List<Dataset> datasets = this.findAllDatasetsByUser(user);
```

Comment:
this call is very slow locally (~17 seconds usually on my machine). i think this is just a local setup issue, though; if you all think not, I can investigate modifying the method to be faster.

Reply:
My local setup shows `3825 ms` for getting all datasets as an admin and `1780 ms` as a chair so I do think this might be a non-issue in deployed environments.

Reply:
Hmm, curious. Maybe I should investigate my local setup's performance - it's an issue on most endpoints that they take a very long time. Using the UI on a local consent is a bit ... slow ....

Reply:
The numbers I quoted are the response times from the swagger page. If you're looking at the UI, there's a whole host of other things going on.